tools/node.d.ts

```diff
@@ -0,0 +1,783 @@
+    braces?: boolean;
+    comments?: boolean | 'all' | 'some' | RegExp;
+    ecma?: ECMA;
+    indent_level?: number;
+    indent_start?: boolean;
+    inline_script?: boolean;
+    ie8?: boolean;
+    keep_quoted_props?: boolean;
+    max_line_len?: boolean;
+    preamble?: string;
+    quote_keys?: boolean;
+    quote_style?: OutputQuoteStyle;
+    safari10?: boolean;
+    semicolons?: boolean;
+    shebang?: boolean;
+    shorthand?: boolean;
+    source_map?: SourceMapOptions;
+    webkit?: boolean;
+    width?: number;
+    wrap_iife?: boolean;
+}
+
+export enum OutputQuoteStyle {
+    PreferDouble = 0,
+    AlwaysSingle = 1,
+    AlwaysDouble = 2,
+    AlwaysOriginal = 3
+}
+
+export interface MinifyOptions {
+    compress?: boolean | CompressOptions;
+    ecma?: ECMA;
+    ie8?: boolean;
+    keep_classnames?: boolean | RegExp;
+    keep_fnames?: boolean | RegExp;
+    mangle?: boolean | MangleOptions;
+    module?: boolean;
+    nameCache?: object;
+    output?: OutputOptions;
+    parse?: ParseOptions;
+    safari10?: boolean;
+    sourceMap?: boolean | SourceMapOptions;
+    toplevel?: boolean;
+    warnings?: boolean | 'verbose';
+}
+
+export interface MinifyOutput {
+    ast?: AST_Node;
+    code?: string;
+    error?: Error;
+    map?: string;
+    warnings?: string[];
+}
+
+export interface SourceMapOptions {
+    content?: RawSourceMap;
+    includeSources?: boolean;
+    filename?: string;
+    root?: string;
+    url?: string | 'inline';
+}
+
+declare function parse(text: string, options?: ParseOptions): AST_Node;
+
+export class TreeWalker {
+    constructor(callback: (node: AST_Node, descend: boolean) => any);
+    directives: object;
+    find_parent(type: AST_Node): AST_Node | undefined;
+    has_directive(type: string): boolean;
+    loopcontrol_target(node: AST_Node): AST_Node | undefined;
+    parent(n: number): AST_Node | undefined;
+    pop(): void;
+    push(node: AST_Node): void;
+    self(): AST_Node | undefined;
+    stack: AST_Node[];
+    visit: (node: AST_Node, descend: boolean) => any;
+}
+
+export class TreeTransformer extends TreeWalker {
+    constructor(before: (node: AST_Node) => AST_Node, after?: (node: AST_Node) => AST_Node);
+    before: (node: AST_Node) => AST_Node;
+    after?: (node: AST_Node) => AST_Node;
+}
+
+export function push_uniq<T>(array: T[], el: T): void;
+
+type DictEachCallback = (val: any, key: string) => any;
+
+export class Dictionary {
+    static fromObject(obj: object): Dictionary;
+    add(key: string, val: any): this;
+    clone(): Dictionary;
+    del(key: string): this;
+    each(fn: DictEachCallback): void;
+    get(key: string): any;
+    has(key: string): boolean;
+    map(fn: DictEachCallback): any[];
+    set(key: string, val: any): this;
+    size(): number;
+}
+
+export function minify(files: string | string[] | { [file: string]: string } | AST_Node, options?: MinifyOptions): MinifyOutput;
+
+export class AST_Node {
+    constructor(props?: object);
```

Comment:
Could you please add this regular (non-static) method to AST_Node?
```
    print_to_string(options?: OutputOptions);
```
It's useful for debugging AST transforms.